tests/utils/test_pidlock.py

```diff
@@ -1,164 +1,181 @@
 """Tests for the PidFileLock utility."""
 
+import os
 import errno
+import pytest
 import logging
-import os
-import shutil
 from pathlib import Path
-from unittest.mock import Mock, patch
-
-import pytest
-
-# Need to mock atexit.unregister, which is called inside PidFileLock.release(). This is crucial for isolated testing.
-import atexit 
-
 from docbuild.utils.pidlock import PidFileLock
 
-log = logging.getLogger(__name__)
-
-# --- Helper Fixtures and Mocks ---
 
 @pytest.fixture
-def mock_pid_running(monkeypatch):
-    """
-    Mocks os.kill to simulate a running process (PID exists).
-    os.kill(pid, 0) succeeds if the process exists.
-    """
-    def mock_kill(pid, sig):
-        if pid == 12345:  # Mock PID of a running process used in the test
-            return
-        # For any other PID, assume dead (simulating ESRCH)
-        raise OSError(errno.ESRCH, "No such process")
-    
-    monkeypatch.setattr(os, "kill", mock_kill)
+def lock_setup(tmp_path):
+    resource_file = tmp_path / "env.production.toml"
+    resource_file.write_text("[dummy]\nkey=value")
+    lock_dir = tmp_path / "locks"
+    return resource_file, lock_dir
 
-@pytest.fixture
-def mock_pid_dead(monkeypatch):
-    """
-    Mocks os.kill to simulate a dead process (PID does not exist).
-    """
-    def mock_kill(pid, sig):
-        # Always raise ESRCH (No such process) for any checked PID
-        raise OSError(errno.ESRCH, "No such process")
-        
-    monkeypatch.setattr(os, "kill", mock_kill)
 
+def test_acquire_and_release_creates_lock_file(lock_setup):
+    resource_path, lock_dir = lock_setup
+    lock_dir.mkdir()
+    lock = PidFileLock(resource_path, lock_dir)
 
-@pytest.fixture
-def lock_setup(tmp_path: Path) -> tuple[Path, Path]:
-    """Provides a resource path and the temporary lock directory."""
-    resource_path = tmp_path / "env.production.toml"
-    resource_path.touch()  # The resource path must exist
-    lock_dir = tmp_path / "locks"
-    return resource_path, lock_dir
+    # Acquire lock
+    lock.acquire()
+    assert lock.lock_path.exists()
+    assert lock._lock_acquired
```

Comment:
I think it's safer to do that in a `with` block:


```suggestion
    with lock:
        assert lock.lock_path.exists()
        assert lock._lock_acquired
```